core/tests/ts-integration/src/modifiers/balance-checker.ts

```diff
@@ -194,43 +196,26 @@ class ShouldChangeBalance extends MatcherModifier {
  *
  * @param receipt Receipt of the transaction to extract fee from.
  * @param from Optional substitute to `receipt.from`.
+ * @param wallet Optional wallet if the api mode is modern
  * @returns Extracted fee
  */
-export function extractFee(receipt: zksync.types.TransactionReceipt, from?: string): Fee {
-    from = from ?? receipt.from;
-
-    const systemAccountAddress = '0x0000000000000000000000000000000000000000000000000000000000008001';
-    // We need to pad address to represent 256-bit value.
-    const fromAccountAddress = ethers.utils.hexZeroPad(ethers.utils.arrayify(from), 32);
-    // Fee log is one that sends money to the system contract account.
-    const feeLog = receipt.logs.find((log) => {
-        return log.topics.length == 3 && log.topics[1] == fromAccountAddress && log.topics[2] == systemAccountAddress;
-    });
-    if (!feeLog) {
+export async function extractFee(receipt: zksync.types.TransactionReceipt, wallet: zksync.Wallet): Promise<Fee> {
+    const trace = await wallet!.provider.send('debug_traceTransaction', [receipt.transactionHash]);
+
+    if (!trace) {
```

Comment:
why do we need this check? Having the `extractFee` as async function is not bad, but most of the diff in tests comes from it, so just curious

Reply:
With this method we are checking whether all the balances were updated correctly(along with fees) and since in this new mode we can't just ask for receipt for native transfers - this is the only way to get fee